extensions/sql-migration/src/api/utils.ts

```diff
@@ -261,3 +256,545 @@ export function clearDialogMessage(dialog: window.Dialog): void {
+	try {
+		accounts = await azdata.accounts.getAllAccounts();
+	} catch (e) {
+		console.log(e);
+		accounts = [];
+	}
+	return accounts;
+}
+
+export async function getAzureAccountsDropdownValues(accounts: azdata.Account[]): Promise<azdata.CategoryValue[]> {
+	let accountsValues: azdata.CategoryValue[] = [];
+	accounts.forEach((account) => {
+		accountsValues.push({
+			name: account.displayInfo.userId,
+			displayName: account.isStale
+				? constants.ACCOUNT_CREDENTIALS_REFRESH(account.displayInfo.displayName)
+				: account.displayInfo.displayName
+		});
+	});
+	if (accountsValues.length === 0) {
+		accountsValues = [
+			{
+				displayName: constants.ACCOUNT_SELECTION_PAGE_NO_LINKED_ACCOUNTS_ERROR,	//
+				name: ''
+			}
+		];
+	}
+	return accountsValues;
+}
+
+export async function getAzureTenants(account?: azdata.Account): Promise<Tenant[]> {
+	let tenants: Tenant[] = [];
+	try {
+		if (account) {
+			tenants = account.properties.tenants;
+		}
+	} catch (e) {
+		console.log(e);
+		tenants = [];
+	}
+	return tenants;
+}
+
+export async function getAzureTenantsDropdownValues(tenants: Tenant[]): Promise<azdata.CategoryValue[]> {
+	let tenantsValues: azdata.CategoryValue[] = [];
+	tenants.forEach((tenant) => {
+		tenantsValues.push({
+			name: tenant.id,
+			displayName: tenant.displayName
+		});
+	});
+	if (tenantsValues.length === 0) {
+		tenantsValues = [
+			{
+				displayName: constants.ACCOUNT_SELECTION_PAGE_NO_LINKED_ACCOUNTS_ERROR,	//
+				name: ''
+			}
+		];
+	}
+	return tenantsValues;
+}
+
+export async function getAzureSubscriptions(account?: azdata.Account): Promise<azureResource.AzureResourceSubscription[]> {
+	let subscriptions: azureResource.AzureResourceSubscription[] = [];
+	try {
+		if (account) {
+			subscriptions = !account.isStale ? await getSubscriptions(account) : [];
+		}
+	} catch (e) {
+		console.log(e);
+		subscriptions = [];
+	}
+	subscriptions.sort((a, b) => a.name.localeCompare(b.name));
+	return subscriptions;
+}
+
+export async function getAzureSubscriptionsDropdownValues(subscriptions: azureResource.AzureResourceSubscription[]): Promise<azdata.CategoryValue[]> {
+	let subscriptionsValues: azdata.CategoryValue[] = [];
+	subscriptions.forEach((subscription) => {
+		subscriptionsValues.push({
+			name: subscription.id,
+			displayName: `${subscription.name} - ${subscription.id}`
+		});
+	});
+	if (subscriptionsValues.length === 0) {
+		subscriptionsValues = [
+			{
+				displayName: constants.NO_SUBSCRIPTIONS_FOUND,
+				name: ''
+			}
+		];
+	}
+	return subscriptionsValues;
+}
+
+export enum SelectableResourceType {
+	ManagedInstance,
+	VirtualMachine,
+	StorageAccount,
+	SqlMigrationService,
+}
+
+export async function getAzureLocations(account?: azdata.Account, subscription?: azureResource.AzureResourceSubscription, resourceType?: SelectableResourceType, resources?: azureResource.AzureSqlManagedInstance[] | SqlVMServer[] | StorageAccount[] | SqlMigrationService[]): Promise<azureResource.AzureLocation[]> {
```

Comment:
please break this function up into typed get calls.  this pattern is hard to maintain